pkg/apis/kops/v1alpha2/cluster.go

```diff
@@ -53,9 +54,11 @@ type ClusterSpec struct {
 	// This might be different that the location when the cluster spec itself is stored,
 	// both because this must be accessible to the cluster,
 	// and because it might be on a different cloud or storage system (etcd vs S3)
-	ConfigBase string `json:"configBase,omitempty"`
+	ConfigBase    string                 `json:"configBase,omitempty"`
+	CloudProvider kops.CloudProviderSpec `json:"-"`
```

Comment:
I think we will need to give it a json representation (once there is new state in here), as otherwise it may not round-trip.  Thinking about this, this is pretty important, because we need to figure out what the representation will be, and then I guess we need to consider whether that representation is "bad enough" to warrant a breaking change.

Reply:
When we add a new field in here we will either have to add manual conversion code or find a place to put a v1alpha2 `CloudProviderSpec` with the existing fields suppressed. The tricky thing about the latter approach is suppressing serialization of the new struct when there are no non-empty, non-suppressed members.